intents.go

```diff
@@ -321,12 +320,27 @@ func unmarshalTransactional(data json.RawMessage) (Transactional, error) {
 			return nil, err
 		}
 		return stake, nil
-	case SupplyType:
-		var supply Supply
-		return supply, json.Unmarshal(data, &supply)
-	case WithdrawSupplyType:
-		var withdraw WithdrawSupply
+	case LoanType:
+		// we use the LoanType for both LoanTo and LoanFrom
+		// the only difference is the Amount field present in LoanFrom,
+		// we can use that to determine the struct to unmarshal into
+
+		var typeDetectStruct struct {
+			Amount string `json:"amount"`
+		}
+
```

Comment:
what typeDetectStruct used for? Maybe a better name?

Reply:
no longer needed since we merged both types into one